reg/drone/phy/kinematics/trans/LinearCovTs.0.1.uavcan

```diff
@@ -0,0 +1,10 @@
+# This is a structural subtype of LinearTs.
```

Comment:
It doesn't make sense to name this Cov, because the Covariance is between two different random variables, and in this case you are just passing the variances (which is specific to the random variable).

Reply:
That is true but the objective here is to follow a consistent convention. We need a generic term that is applicable both to covariance and error variance, I figured that `Cov` is an appropriate choice. Do you have a better idea that would work in both cases?

Reply:
I don't. I would say that or we propagate the covariances (which increases the data type size), or we just send the variances, which in most of the cases is enough.

Reply:
The quantities contained in this message are not correlated, so if we included the covariance it would be a (nearly) diagonal matrix anyway. The question is, can we accept `Cov` as a convention even if we're not really dealing with covariance?

Reply:
> The question is, can we accept Cov as a convention even if we're not really dealing with covariance?

Short answer, no. And doing it will only generate confusion. Or you pass the URT of the covariance matrix or you just pass the variances (which will be the diagonal of the covariance matrix).

The other thing I am seeing now: why is `Linear` bidimensional and not tridimensional?

Reply:
> Short answer, no. 

Do you have a specific suggestion?

> The other thing I am seeing now: why is Linear bidimensional and not tridimensional?

It is one-dimensional because it is only used to represent one-dimensional kinetic states. Multidimensional geometry will be supported later under the existing framework.

Reply:
> It is one-dimensional because it is only used to represent one-dimensional kinetic states. Multidimensional geometry will be supported later under the existing framework.

One dimension means that, for example, for position, you have the x variance OR the y variance OR the z variance.

Reply:
That is a very astute observation. Let's have a call, this is going nowhere.

Reply:
I would say that using `cov` won't create any ambiguity: the matrix containing both variances and covariances is called "covariance matrix" and the variance is just a special name when both random variables are the same.